public/app/plugins/datasource/loki/querySplitting.ts

```diff
@@ -1,14 +1,30 @@
 import { Subscriber, map, Observable } from 'rxjs';
 
-import { DataQueryRequest, DataQueryResponse } from '@grafana/data';
+import { DataQueryRequest, DataQueryResponse, dateTime, TimeRange } from '@grafana/data';
 import { LoadingState } from '@grafana/schema';
 
 import { LokiDatasource } from './datasource';
-import { partitionTimeRange, combineResponses, resultLimitReached } from './queryUtils';
+import { getRanges } from './metricTimeSplit';
+import { combineResponses, resultLimitReached } from './queryUtils';
 import { LokiQuery } from './types';
 
 export function runPartitionedQuery(datasource: LokiDatasource, request: DataQueryRequest<LokiQuery>) {
-  const partition = partitionTimeRange(request.range);
+  // we currently assume we are only running metric queries here.
+  // for logs-queries we will have to use a different time-range-split algorithm.
+  const partition: TimeRange[] = getRanges(
```

Comment:
Could you please move this to another version of `partitionTimeRange()`, so we get timeRange instances here and don't have to create them? It's currently a bit leaky.

And secondarily, could you please test if we need something like `PARTITION_LIMIT`? Problem: in a query that should be split, select `1 year` as the time range. In the previous implementation it would generate a ~ infinite loop which would halt the browser.

Reply:
i extract that part into a separate `partitionTimeRange()` function, i hope it's suitable (if not we can work on it more, no problem).

(will handle the second comment in a separate comment)

Reply:
i also added the partition-limit feature.